common/app/common/LinkTo.scala

```diff
@@ -49,18 +50,20 @@ trait LinkTo extends Logging {
     case RssPath(path, format) =>
       ProcessedUrl(urlFor(path, edition) + format)
     case AmpPath(path, format) =>
-      ProcessedUrl(urlFor(path, edition, secure = true) + format)
+      ProcessedUrl(urlFor(path, edition, secure = true, isAmp = true) + format)
     case GuardianUrl(_, path) =>
       ProcessedUrl(urlFor(path, edition))
     case otherUrl =>
       ProcessedUrl(otherUrl, true)
   }
 
-  private def urlFor(path: String, edition: Edition, secure: Boolean = false): String = {
+  private def urlFor(path: String, edition: Edition, secure: Boolean = false, isAmp: Boolean = false): String = {
```

Comment:
It feels like have an argument of `urlHost: Option[String] = None` might be more generic and then you could have something like:
```scala
val url = s"${urlHost.getOrElse(host)}/$editionalisedPath"
```

Just a thought, perhaps I'm wrong.